developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -129,132 +101,308 @@ Once it has been instantiated, you will notice that the client API is different
+
+#### Better Filter System
+
+We've made it easier to use filters with conditions by adding a `Filter` helper class. The overall experience of filters has been streamlined for cleaner more concise development. 
+
+<Tabs groupId="languages">
+<TabItem value="jsv3" label="JS/TS (v3)">
+
+```ts
+import weaviate, { Filters } from 'weaviate-client';
+const myCollection = client.collections.get('JeopardyQuestion');
+     
+const result = await myCollection.query.fetchObjects({
+  returnProperties: ['question', 'answer','round', 'points'],
+  filters: Filters.and(
+     myCollection.filter.byProperty('round').equal('Double Jeopardy!'),
+     myCollection.filter.byProperty('points').lessThan(600)
+    ),
+  limit: 3,
+ })
+
+console.log(JSON.stringify(result, null, 2));
+```
+
+</TabItem>
+<TabItem value="jsv2" label="JS/TS (v2)">
+
+```ts
+result = await client.graphql
+  .get()
+  .withClassName('JeopardyQuestion')
+  .withWhere({
+    operator: 'And',
+    operands: [
+      {
+        path: ['round'],
+        operator: 'Equal',
+        valueText: 'Double Jeopardy!',
+      },
+      {
+        path: ['points'],
+        operator: 'LessThan',
+        valueInt: 600,
+      },
+    ],
+  })
+  .withLimit(3)
+  .withFields('question answer round points')
+  .do();
+
+console.log(JSON.stringify(result, null, 2));
+```
+
+</TabItem>
+</Tabs>
+
+#### New Generate Namespace
+
+To perform generative queries, we have included a new `generate` namespace in the `v3` client. Through this namespace you have make queries as you would with the `query` namespace. 
+
+<Tabs groupId="languages">
+<TabItem value="jsv3" label="JS/TS (v3)">
+
+```ts
+const generatePrompt = `Convert this quiz question: {question} and answer: {answer} into a trivia tweet.`;
+
+const myCollection = client.collections.get('JeopardyQuestion');
+const result = await myCollection.generate.nearText(['World history'],{
+    singlePrompt: generatePrompt,
+  },{
+    limit: 2,
+    returnProperties: ['round'],
+})
+
+console.log(JSON.stringify(result.objects, null, 2));
+```
 
-#### InsertMany
-#### Migration?
-#### Client.close()
-#### Filters
-#### Named vectors? 
-#### cleaner return object 
-#### Error handling 
+
+</TabItem>
+<TabItem value="jsv2" label="JS/TS (v2)">
+
+```ts
+generatePrompt = 'Convert this quiz question: {question} and answer: {answer} into a trivia tweet.';
+
+result = await client.graphql
+  .get()
+  .withClassName('JeopardyQuestion')
+  .withGenerate({
+    singlePrompt: generatePrompt,
+  })
+  .withNearText({
+    concepts: ['World history'],
+  })
+  .withFields('round')
+  .withLimit(2)
+  .do();
+
+console.log(JSON.stringify(result, null, 2));
+```
+
+</TabItem>
+</Tabs>
+
+#### Cleaner Return object 
+
+The new client has made its return object cleaner by making it easier to access important information like your UUID, results of generative queries and object metadata. 
```

Comment:
Consider:
The new client has a cleaner return object. It is easier to access important information like object UUIDs, object metadata, and generative query results.